src/webui/api/torrentscontroller.h

```diff
@@ -91,4 +95,10 @@ private slots:
     void exportAction();
     void SSLParametersAction();
     void setSSLParametersAction();
+    void metadataAction();
+
+private:
+    void onMetadataDownloaded(const BitTorrent::TorrentInfo &metadata);
+
+    QHash<BitTorrent::InfoHash, BitTorrent::TorrentInfo> m_torrentMetadata;
```

Comment:
Retrieved metadata is stored in the session and is only cleaned up when the session is destroyed. Alternatively, we could clean up the metadata once it's retrieved via the API.

Reply:
>Alternatively, we could clean up the metadata once it's retrieved via the API.

:+1:
I believe the metadata won't be needed more than once in the vast majority of cases.

Reply:
Given that the `/add` API now supports using this metadata, the metadata will be deleted when the torrent is added.